zummon/src/commonMain/kotlin/companysurvey/Validation.kt

```diff
@@ -16,231 +16,288 @@ enum class Status {
+            if (kleinOrGroot == KleinverbruikOrGrootverbruik.KLEINVERBRUIK) {
+                results.add(validateKleinverbruik(electricity))
             }
         }
-        KleinverbruikOrGrootverbruik.KLEINVERBRUIK -> {
-            val connectionCapacity = gridConnection.electricity.getPhysicalConnectionCapacityKw()
 
-            if (connectionCapacity == null) {
-                ValidationResult(Status.MISSING_DATA, "No connection capacity data for Small consumption")
-            } else if (connectionCapacity <= KleinverbruikElectricityConnectionCapacity._3x80A.toKw()) {
-                ValidationResult(Status.VALID, "Small consumption connection capacity is valid")
+        return results
+    }
+
+    fun validateGrootverbruik(grootverbruik: CompanyGrootverbruik?): List<ValidationResult> {
+        val results = mutableListOf<ValidationResult>()
+
+        if (grootverbruik == null) {
+            results.add(ValidationResult(Status.INVALID, "Large consumption data is not provided"))
+        } else {
+            grootverbruik.contractedConnectionDeliveryCapacity_kW ?: results.add(ValidationResult(Status.MISSING_DATA, "Connection delivery capacity is not provided for large consumption"))
+            grootverbruik.physicalCapacityKw ?: results.add(ValidationResult(Status.MISSING_DATA, "Physical Capacity Kw is not provided for large consumption"))
+            grootverbruik.contractedConnectionFeedInCapacity_kW ?: results.add(ValidationResult(Status.MISSING_DATA, "Connection delivery capacity is not provided for large consumption"))
+        }
+
+        return results
+    }
+
+    // Validator for grootverbruik physical connection > 3x80A (55.2 kW)
+    fun validateGrootverbruikPhysicalCapacity(electricity: Electricity): ValidationResult {
+        val connectionCapacity = electricity.getPhysicalConnectionCapacityKw()
+
+        return if (connectionCapacity == null) {
+            ValidationResult(Status.MISSING_DATA, "No physical capacity data for Large Consumption")
+        } else if (connectionCapacity > KleinverbruikElectricityConnectionCapacity._3x80A.toKw()) {
+            ValidationResult(Status.VALID, "Large Consumption physical capacity is valid")
+        } else {
+            ValidationResult(Status.INVALID, "Large Consumption physical capacity ${connectionCapacity} is below 3x80A")
+        }
+    }
+
+    fun validateKleinverbruik(electricity: Electricity): ValidationResult {
+        return if (electricity.kleinverbruikOrGrootverbruik == KleinverbruikOrGrootverbruik.KLEINVERBRUIK) {
+             if (electricity.kleinverbruik == null) {
+                ValidationResult(Status.INVALID, "Small consumption data is not provided")
+            } else if (electricity.kleinverbruik.connectionCapacity != null) {
+                // Validate that kleinverbruik connection capacity <= 3x80A
+                if (electricity.kleinverbruik.connectionCapacity > KleinverbruikElectricityConnectionCapacity._3x80A) {
```

Comment:
This is always false but I'll stop nagging about it :smile:

Reply:
I think I flip it to make us both happy